field_encrypt.module

```diff
@@ -131,44 +124,115 @@ function field_encrypt_form_field_add_form_builder($entity_type, \Drupal\field\E
     /**
      * @var $field_encrypt_process_entities \Drupal\field_encrypt\FieldEncryptProcessEntities
      */
-    $field_encrypt_process_entities = \Drupal::service('field_encrypt.process_entities');
-    if ($form_encryption === 1) {
-      $field_encrypt_process_entities->encryptStoredField($field_entity_type, $field_name);
-    }
-    elseif ($form_encryption === 0) {
-      $field_encrypt_process_entities->decryptStoredField($field_entity_type, $field_name);
-    }
+    // @TODO: refactor logic for re-encrypting existing fields - currently broken, so commented out.
+    //    $field_encrypt_process_entities = \Drupal::service('field_encrypt.process_entities');
+    //    if ($form_encryption === 1) {
+    //      $field_encrypt_process_entities->encryptStoredField($field_entity_type, $field_name);
+    //    }
+    //    elseif ($form_encryption === 0) {
+    //      $field_encrypt_process_entities->decryptStoredField($field_entity_type, $field_name);
+    //    }
   }
 }
 
 /**
- * Encrypt fields before they are saved.
+ * Implements hook_entity_insert().
+ *
+ * Encrypts the entity after being saved for the first time.
+ */
+function field_encrypt_entity_insert(Drupal\Core\Entity\EntityInterface $entity) {
+  // Save the entity again, but now encrypted.
+  // @TODO: check if there's not a cleaner way to do this.
+  $saved_entity = entity_load($entity->getEntityTypeId(), $entity->id());
+  field_encrypt_entity_encrypt($saved_entity);
+}
+
+/**
+ * Implements hook_entity_update().
+ *
+ * Encrypts the entity after being updated.
+ */
+function field_encrypt_entity_update(Drupal\Core\Entity\EntityInterface $entity) {
+  field_encrypt_entity_encrypt($entity);
+}
+
+/**
+ * Encrypt the saved entity.
  *
  * @param \Drupal\Core\Entity\EntityInterface $entity
  */
-function field_encrypt_entity_presave(\Drupal\Core\Entity\EntityInterface $entity) {
-  if (!($entity instanceof Drupal\Core\Entity\ContentEntityInterface)) {return;}
-  /**
-   * @var $field_encrypt_process_entities \Drupal\field_encrypt\FieldEncryptProcessEntities
-   */
+function field_encrypt_entity_encrypt(Drupal\Core\Entity\EntityInterface $entity) {
+  if (($entity instanceof Drupal\field_encrypt\Entity\EncryptedFieldValueInterface) || !($entity instanceof Drupal\Core\Entity\ContentEntityInterface)) {
```

Comment:
It makes sure to only encrypt ContentEntities, not ConfigEntities.
Not sure if the code would work for ConfigEntities though, and if that would desired behaviour...

I also check if the entity being inserted / updated isn't a EncryptedFieldValue entity, since there seems little reason to encrypt the entity that stores the encrypted data. So it prevents the FieldEncryptProcessEntities service from being needlessly called and doing checks, if we already know it won't be necessary.